### PR TITLE
Zero-shot Computer Vision - Draft Introduction 

### PR DESCRIPTION
**Pull request by mmhamdy**
The first draft of the introduction chapter to zero-shot computer vision.

Addressing the issue #43

### CODE REVIEW

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+[comment]: # (TODO: Diagram showing the categories of ZSL methods and the sub-categories)
+
+The choice between embedding-based and generative-based approaches, like every other choice that you have to make in machine learning, *depends* on the specifics of the task at hand and the available resources. Embedding-based methods are often preferred for their efficiency and scalability, while generative-based methods offer greater flexibility and potential for handling complex data.
+
+But Anyway, In this unit we are going to address *only* embedding-based methods, including **attention-based embedding-based methods**. Zero-shot learning is a big topic and comprehensive coverage may require a course of its own. Interested readers may check the provided additional readings and have fun.
```
Comment (ricard-inho) [inline chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:93]: > But anyway, in this

**Review by ricard-inho — COMMENTED** [review]
Great introduction. Looking forward to the images but looks great on my end.

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx [hunk 2]
```diff
@@ -0,0 +1,110 @@
+# Introduction
+
+Besides fundamentals, this unit assumes familiarity with concepts in transfer and multimodal learning. If you are not viewing the course sequentially in the provided order, we suggest at least reading the transfer learning section in unit 2, and the whole of unit 4.
```
Comment (merveenoyan) [inline chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:3]: Maybe we could have this in Tip/Warning blocks (you can check out readme for how to do it)

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+Let’s begin by answering a simple question, *how can we humans do it? How can we recognize objects that we have not seen before?*
+
+Yes, you’re absolutely right! We need some **other information** about that object. We can easily identify a tiger, even if we haven’t seen one before, given that we know what a lion looks like. A tiger is a lion with stripes and minus the mane. A zebra is a black-and-white striped version of a horse. Deadpool is a red-black version of Spiderman.
+
+As we can see, zero-shot computer vision is essentially multi-modal. If generalization is like learning a language by reading books and talking to people. Zero-shot computer vision is like learning a language by reading a dictionary and listening to someone describe what it sounds like.
```
Comment (merveenoyan) [inline chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:56]:
```suggestion
As we can see, zero-shot computer vision is essentially multi-modal. If generalization is like learning a language by reading books and talking to people, zero-shot computer vision is like learning a language by reading a dictionary and listening to someone describe what it sounds like.
```

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+Can we think of zero-shot learning as an unsupervised learning problem? No, this is still supervised learning. In unsupervised learning, the model learns from **unlabeled data**. In zero-shot learning, the model learns from **dataless labels**, remember it was called **dataless classification** in its infancy.
+
+### What Other Information?
```
Comment (merveenoyan) [inline chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:60]: Can you elaborate what you mean with this header?

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+It is called **Semantic or Auxiliary Information**. This auxiliary information provides a semantic bridge between the visual features and the unseen classes. By incorporating this multi-modal information (Text & Image), zero-shot computer vision models can learn to recognize objects even if they have never seen them before visually. In other words, semantic information embeds **both** seen and unseen classes in high-dimensional vectors, and it comes in many different forms:
```
Comment (merveenoyan) [inline chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:62]: What is called **semantic or auxiliary information**?

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+1. **Attribute vectors**: Thinks of attributes as a tabular representation of the different features of the object. 
+2. **Textual Descriptions**: A text describing the object contained in the image similar to image captions.
+3. **Class label vectors**: Simply by using the embeddings of class labels.
```
Comment (merveenoyan) [inline chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:66]: The above ones are describing "what", but this one is describing "how" which causes a bit of a confusion

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+Using this semantic information, you train a model to learn a mapping function between the image features and the semantic features. At inference time, the model predicts the class label by looking for the closest label in the semantic space by using, for example, k-nearest neighbor. We can say that we are transferring knowledge obtained from seen classes with the help of semantic information.
+
+Different zero-shot computer vision methods differ in what **semantic information** they use and which **embedding space** they utilize at inference.
+
+[comment]: # (TODO: Illustration showing the pipeline of zero-shot learning)
+
+### How Is This Different from Transfer Learning?
+
+Good question! Zero-shot learning (ZSL) falls within the broader category of transfer learning, specifically under the umbrella of **heterogeneous transfer learning**. This means that ZSL relies on transferring knowledge gained from a **source domain** (seen classes) to a *distinct* **target domain** (unseen classes).
```
Comment (merveenoyan) [inline chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:78]:
```suggestion
Good question! Zero-shot learning (ZSL) falls within the broader category of transfer learning, specifically under the umbrella of **heterogeneous transfer learning**. This means that ZSL relies on transferring knowledge gained from a **source domain** (seen classes) to a **distinct target domain** (unseen classes). 
```

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx (hunk 1 above)
Comment (merveenoyan) [inline chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:93]:
```suggestion
But anyway, in this unit we will address *only* embedding-based methods, including **attention-based embedding-based methods**. Zero-shot learning is a big topic and comprehensive coverage may require a standalone course. Interested readers may check the provided additional readings and have fun.
```

**Review by merveenoyan — COMMENTED** [review]
Loved the introduction, left some comments but otherwise looks great to me 💜

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx (hunk 2 above)
Comment (mmhamdy) [reply chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:3]: Sure, makes sense. I'm just wondering if there's a third option other than Tip/Warning. Note won't work, right?

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+Actually, I wasn’t totally accurate when I said that the model is expected to generalize to all cat pictures that it has not seen. It is expected to generalize to all cat pictures that come from the same distribution as the image data it was trained on. Simply, if you trained your model on cat selfies and then presented it with a cartoon picture of a cat, it probably won’t be able to recognize it. These two pictures come from totally different distributions or domains. Making your cat selfie model able to recognize cartoon cats is referred to as **domain adaptation** (we’ll briefly talk about it later). It's like taking all the knowledge your model learned about real cats and teaching it to recognize their animated cousins.
+
+So, we’ve gone from generalization, recognizing the unseen Alice’s and Ted’s cat pictures, to domain adaptation, recognizing animated cat pictures. But we’re much Greedier than that. You don’t want your model to be able to only recognize your cat pictures, or Alice’s and Ted’s, or not even cartoon cats! Having a model trained on cat pictures, you also want it to recognize pictures of llamas and falcons.
```
[the remark this answers is not in the document]
Comment (mmhamdy) [reply chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:11]: No, you're right, should have been greedier. Will fix it.

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx (hunk 4 above)
Comment (mmhamdy) [reply chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:60]: Sorry! Maybe the preceding paragraph broke the flow. It was supposed to be a question about this previous part:

> Yes, you’re absolutely right! We need some **other information** about that object. We can easily identify a tiger, even if we haven’t seen one before, given that we know what a lion looks like. A tiger is a lion with stripes and minus the mane. A zebra is a black-and-white striped version of a horse. Deadpool is a red-black version of Spiderman.

I think wrapping the preceding paragraph in a tip block would make the flow smoother.

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx (hunk 1 above)
Comment (mmhamdy) [reply chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:93]: That's right. Will fix it.

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx (hunk 5 above)
Comment (mmhamdy) [reply chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:62]: This is an answer to "What other information?", right?! Or is this a question about it? sorry😅, I should've been more clear.

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx (hunk 6 above)
Comment (mmhamdy) [reply chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:66]: Do you mean the class label vectors? I see, the wording does sound like a how. Will come with a better one.

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx (hunk 5 above)
Comment (merveenoyan) [reply chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:66]: Ah no worries, I meant you start the paragraph with "It is called.." what does it refer to in this case?

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx (hunk 2 above)
Comment (merveenoyan) [reply chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:4]: ah there's no other I think

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx (hunk 4 above)
Comment (merveenoyan) [reply chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:64]: Yes, I think the header itself (maybe make it bold?) and following paragraph is a bit breaking the readability

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx (hunk 5 above)
Comment (mmhamdy) [reply chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:66]: Aha! got you. I think I'm going to do some little rewriting for this section.

**Comment by mmhamdy** [comment]
Hi @merveenoyan, I did some rewriting to the section about semantic information. Check it out and tell me what you think.

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+Actually, I wasn’t totally accurate when I said that the model is expected to generalize to all cat pictures that it has not seen. It is expected to generalize to all cat pictures that come from the same distribution as the image data it was trained on. Simply, if you trained your model on cat selfies and then presented it with a cartoon picture of a cat, it probably won’t be able to recognize it. These two pictures come from totally different distributions or domains. Making your cat selfie model able to recognize cartoon cats is referred to as **domain adaptation** (we’ll briefly talk about it later). It's like taking all the knowledge your model learned about real cats and teaching it to recognize their animated cousins.
```
Comment (merveenoyan) [inline chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:11]:
```suggestion
Actually, we weren't right when we said that the model is expected to generalize to all cat pictures that it has not seen. It is expected to generalize to all cat pictures from the same distribution as the image data it was trained on. Simply, if you trained your model on cat selfies and then presented it with a cartoon picture of a cat, it probably won’t be able to recognize it. These two pictures come from totally different distributions or domains. Making your cat selfie model able to recognize cartoon cats is referred to as **domain adaptation** (we’ll briefly talk about it later). It's like taking all the knowledge your model learned about real cats and teaching it to recognize their animated cousins.
```

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+So, we’ve gone from generalization, recognizing the unseen Alice’s and Ted’s cat pictures, to domain adaptation, recognizing animated cat pictures. But we’re much greedier than that. You don’t want your model to be able to only recognize your cat pictures, or Alice’s and Ted’s, or not even cartoon cats! Having a model trained on cat pictures, you also want it to recognize pictures of llamas and falcons.
+
+Well, now we’re on the turf of Zero-shot Learning. Also known as, ZSL.
```
Comment (merveenoyan) [inline chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:15]:
```suggestion
Well, now we’re on the turf of zero-shot learning (also known as ZSL).
```

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+
+[comment]: # (TODO: Illustration showing the difference between generalization, domain adaptation, and ZSL)
```
Comment (merveenoyan) [inline chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:17]: Were you able to accomplish this?

**Review by merveenoyan — COMMENTED** [review]
I did a bit nit-picking, are you going to add the images? If not, we can remove the texts and approve

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx (hunk 11 above)
Comment (mmhamdy) [reply chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:17]: I'm working on this and the other chapters. We can merge the text, the comments won't appear in the rendered version.

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+
+## How Does Zero-shot Learning Work In Computer Vision?
+
+Now that we know what zero-shot learning is, it would be nice to know how it is applied to computer vision, right? This part will be addressed in more detail in the coming chapters, but let’s paint a broad picture here to break the ice.
+
+In NLP, zero-shot learning is (although it wasn’t always the case) pretty straightforward. Many language models are trained on massive datasets of text, learning to predict the next word in a given sequence. This ability to capture the underlying patterns and semantic relationships within language allows these models to perform surprisingly well on tasks they weren’t explicitly trained on. A nice example is GPT-2 being able to summarize a passage when TL;DR is appended to the prompt. Zero-shot computer vision is another story.
+
+Let’s begin by answering a simple question, *how can we humans do it? How can we recognize objects that we have not seen before?*
```
Comment (MKhalusova) [inline chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:54]:
```suggestion
Let’s begin by answering a simple question, *how can we humans do it? How can we recognize objects that we have not seen before?
```

**Review by MKhalusova — APPROVED** [review]
Excellent work!

File: chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx (hunk 12 above)
Comment (mmhamdy) [reply chapters/en/Unit 11  - Zero Shot Computer Vision/1.mdx:54]: Hi, @MKhalusova, are you suggesting the removal of italicization?

**Comment by merveenoyan** [comment]
@mmhamdy if you can add the image and commit @MKhalusova I'll give a one last review and we can merge (I'll likely have no other comments)

**Comment by mmhamdy** [comment]
Hi @merveenoyan , I'm still working on the images. If possible, we can merge this and I'll address the five images in the next PR.

**Review by merveenoyan — APPROVED** [review]

**Pull request merged**